test/unit/event/send-acknowledgement-error-event.test.js

```diff
@@ -0,0 +1,72 @@
+jest.mock('../../../app/event/send-processing-ack-error-event')
+const sendProcessingAckErrorEvent = require('../../../app/event/send-processing-ack-error-event')
+
+jest.mock('../../../app/event/send-processing-ack-invalid-bank-details-error-event')
+const sendProcessingAckInvalidBankDetailsErrorEvent = require('../../../app/event/send-processing-ack-invalid-bank-details-error-event')
+
+const sendAcknowledgementErrorEvent = require('../../../app/event/send-acknowledgement-error-event')
+
+const mockFRN = require('../../mock-frn')
+
+let mockAcknowledgementError
+let mockHoldCategoryNameDR
+let mockHoldCategoryNameBAA
+
+describe('send acknowledgement event', () => {
+  beforeEach(() => {
+    mockAcknowledgementError = JSON.parse(JSON.stringify(require('../../mock-acknowledgement-error')))
+    mockHoldCategoryNameDR = JSON.parse(JSON.stringify(require('../../../app/constants/hold-categories-names'))).DAX_REJECTION
+    mockHoldCategoryNameBAA = JSON.parse(JSON.stringify(require('../../../app/constants/hold-categories-names'))).BANK_ACCOUNT_ANOMALY
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+    jest.resetModules()
+  })
+
+  test('should call sendProcessingAckErrorEvent when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementErrorEvent(mockHoldCategoryNameDR, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalled()
+  })
+
+  test('should call sendProcessingAckErrorEvent once when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementErrorEvent(mockHoldCategoryNameDR, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call sendProcessingAckErrorEvent with the unsuccessful ack object when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementErrorEvent(mockHoldCategoryNameDR, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalledWith(mockAcknowledgementError)
+  })
+
+  test('should throw error with "Issue proccessing acknowledgement error event" when sendProcessingAckErrorEvent throws error with "Issue proccessing acknowledgement error event"', async () => {
```

Comment:
Similar to the error checking in `test/unit/acknowledgement/process-invalid.test.js` I would include error tests for:


- the function throws an error (i.e. how you've done it `test/unit/acknowledgement/process-invalid.test.js`)
- the function throws the right error class (i.e. throws Error) => `expect(wrapper).rejects.toThrow(Error)` found in `test/unit/acknowledgement/update-acknowledgement.test.js`